### PR TITLE
Feature/support collapsing whitespaces

### PR DESCRIPTION
**Pull request by vokiput**
Implements code for #3884

### CODE REVIEW

**Review by pekkaklarck — CHANGES_REQUESTED** [review]
Looks great! The only problem is that keywords should get a not that this is a new feature in RF 4.1. Could you please still add them?

I noticed that the PR splits several long list comprehensions and generator expressions in related code to multiple lines. That's nice in general as there are some overly long lines, but it is perhaps done a bit too aggressively. We have 88 character max line length in Python code (inspired by [Black](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length)) and many expressions would fit into it. Too short lines are better that too long lines so you can leave them as-is, but merging not-too-long lines back together is fine too.

**Comment by vokiput** [comment]
1. added notes that this is a new feature in RF 4.1
2. Restored long list comprehensions and generator expressions

File: src/robot/libraries/BuiltIn.py [hunk 1]
```diff
@@ -636,7 +636,7 @@ def should_be_equal(self, first, second, msg=None, values=True,
         | Should Be Equal | ${x} | expected | ignore_case=True | formatter=repr |
 
         ``formatter`` is new in Robot Framework 3.1.2 and ``strip_spaces`` is new
```
Comment (pekkaklarck) [inline src/robot/libraries/BuiltIn.py:638]: "and" should be replaced by a comma here, but we could also remove the reference to old RF 3.1.2 version and the whole problems goes away. I'll do that myself so no changes needed.

**Review by pekkaklarck — APPROVED** [review]

**Comment by pekkaklarck** [comment]
When looking at the code a bit more closely, I noticed that `should_not_contain_any` calls a wrong helper method. I'll fix that and also enhance tests so that the bug is caught.

I also noticed that the long list comprehensions discussed above can be shortened by enhancing `_strip_spaces` and `_collapse_spaces` helpers so that they check is the value a string or not. This makes it possible to change `[helper(x) if is_string(x) else x for x in container]` to `[helper(x) for x in container]`. I'll do this change as well.

**Pull request merged**